azurelinuxagent/common/protocol/imds.py

```diff
@@ -28,8 +27,8 @@
 IMDS_INTERNAL_SERVER_ERROR = 3
 
 
-def get_imds_client():
-    return ImdsClient()
+def get_imds_client(wireserver_endpoint):
```

Comment:
The IMDS client had a dependency on the protocol just to get the endpoint; Instead, now we pass the endpoint as parameter.